response_writer.go

```diff
@@ -61,6 +61,10 @@ func (w *responseWriter) callBefore() {
 }
 
 func (w *responseWriter) WriteHeader(s int) {
+	if w.Written() {
+		return
+	}
```

Comment:
Writing status code twice does not make sense, so ignore subsequent calls of `WriteHeader`.